core/services/relay/evm/mercury_provider.go

```diff
@@ -103,3 +107,37 @@ func (p *mercuryProvider) ContractTransmitter() ocrtypes.ContractTransmitter {
 func (p *mercuryProvider) MercuryServerFetcher() relaymercury.MercuryServerFetcher {
 	return p.transmitter
 }
+
+func (p *mercuryProvider) ChainReader() relaymercury.ChainReader {
+	return p.chainReader
+}
+
+var _ relaymercury.ChainReader = (*chainReader)(nil)
+
+type chainReader struct {
+	tracker httypes.HeadTracker
+}
+
+func NewChainReader(h httypes.HeadTracker) relaymercury.ChainReader {
+	return &chainReader{
+		tracker: h,
+	}
+}
+
+func (r *chainReader) LatestHeads(ctx context.Context, k int) ([]relaymercury.Head, error) {
```

Comment:
This never returns an error

Reply:
Every method in the relay API must return an error